### DISCUSSION TITLE
How to use createInfiniteScroll with a non-number page

### DISCUSSION THREAD

**Discussion by onx2** [solidjs-community/solid-primitives#620] [Q&A]
I'm using GraphQL and the Relay specification which prefers an opaque string cursor instead of a page number. Does anyone have any recommendations on how to implement using the current state or would this utility need to be updated to allow string | number?

**Comment by atk** [comment]
Just use your fetcher function to not use the page.

**Reply by onx2** [reply]
Could you please elaborate? The fetcher function has a hard coded page as a `number` and is used internally to create a `page` signal with that type. Wouldn't it be more flexible to use a generic type for the page and not assume it is a number?

**Reply by atk** [reply]
IIRC, there was already a PR to do that, but it did not support the previous use case.

**Reply by onx2** [reply]
Do you have a suggestion on how to implement the `string` version by doing _"use your fetcher function to not use the page"_? I don't understand what you mean. 

Are you suggesting tracking the page internally and decoding on the FE while passing the string the BE? Ideally this utility is agnostic of pagination approach.

**Comment by atk** [comment]
If it is not numeric, you will need to fetch the string id for the subsequent pages from the server in any case. Why not do so directly in your fetcher. Nobody tells you to even use the number. Here's our example adapted to that use case:

```ts
const [nextPageId, setNextPageId] = createSignal('');
async function fetcher() {
  let elements: string[] = [];
  let res = await fetch(`/api/page/${nextPageId()}`);
  if (res.ok) {
    let json = await res.json();
    // get the id from the next page from the data:
    setNextPageId(json.nextPageId);
    json.docs.forEach((b: any) => elements.push(b.title));
  }
  return elements;
}
```
